src/Microsoft.AspNetCore.OData/Extensions/Endpoint/ODataEndpointLinkGenerator.cs

```diff
@@ -0,0 +1,135 @@
+        }
+
+        /// <inheritdoc/>
+        public override string GetUriByAddress<TAddress>(HttpContext httpContext, TAddress address, RouteValueDictionary values, RouteValueDictionary ambientValues = null, string scheme = null, HostString? host = null, PathString? pathBase = null, FragmentString fragment = default, LinkOptions options = null)
+        {
+            return _innerGenerator.GetUriByAddress(httpContext, address, values, ambientValues, scheme, host, pathBase, fragment, options);
+        }
+
+        /// <inheritdoc/>
+        public override string GetUriByAddress<TAddress>(TAddress address, RouteValueDictionary values, string scheme, HostString host, PathString pathBase = default, FragmentString fragment = default, LinkOptions options = null)
+        {
+            return _innerGenerator.GetUriByAddress(address, values, scheme, host, pathBase, fragment, options);
+        }
+
+        // Noted: simple workaround to bind the value for the prefix template
+        // Shold replace it using the standard ASP.NET Core way later.
+        private static string BindPrefixTemplate(string prefix, RouteValueDictionary values, RouteValueDictionary ambientValues)
+        {
+            IList<string> templates = new List<string>();
+            int startIndex = 0;
+            while (true)
+            {
+                int start = prefix.IndexOf('{', startIndex);
+                int end = prefix.IndexOf('}', start + 1); // start == -1 is ok
+                if (start == -1 || end == -1)
+                {
+                    break;
+                }
+
+                startIndex = end + 1;
+
+                string subStr = prefix.Substring(start, end - start + 1);
+                templates.Add(subStr);
+            }
+
+            foreach (var item in templates)
+            {
+                string variable = item.Substring(1, item.Length - 2); // remove { and }
+
+                if (values != null && values.TryGetValue(variable, out object valueFromValues))
+                {
+                    prefix = prefix.Replace(item, valueFromValues as string);
+                }
+                else if (ambientValues != null && ambientValues.TryGetValue(variable, out object valueFromAmbient))
+                {
+                    prefix = prefix.Replace(item, valueFromAmbient as string);
+                }
+            }
+
+            return prefix;
+        }
+
+        private static string CombinePathSegments(string routePrefix, string odataPath)
+        {
+            if (String.IsNullOrEmpty(routePrefix))
+            {
+                return odataPath;
+            }
+            else
+            {
+                return String.IsNullOrEmpty(odataPath) ? routePrefix : routePrefix + '/' + odataPath;
+            }
+        }
+
+        private static string UriEncode(string str)
+        {
+            string escape = Uri.EscapeUriString(str);
+            escape = escape.Replace("#", _escapedHashMark);
```

Comment:
>"#" [](start = 36, length = 3)

Declare these strings as constants? It may be better to use the char overload instead of the string overload for better performance.

Reply:
Replace doesn't have a overload to replace a char in a string. and for better readability, I'd like to use "#" directly here. The LinkGenerator mayeb change later. 

---
In reply to: [376006516](https://github.com/OData/WebApi/pull/2035#discussion_r376006516) [](ancestors = 376006516)